src/accelerate/data_loader.py

```diff
@@ -304,7 +304,10 @@ def __iter__(self):
         self.gradient_state._set_end_of_dataloader(False)
         dataloader_iter = super().__iter__()
         # We iterate one batch ahead to check when we are at the end
-        current_batch = next(dataloader_iter)
+        try:
+            current_batch = next(dataloader_iter)
+        except StopIteration:
+            raise StopIteration("Tried iterating over an empty dataloader")
```

Comment:
No we shouldn't raise here, just yield nothing. A training loop over an empty dataloader works, just doesn't do anything.
We can put a warning however, just in case the user didn't intend that.